sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/http/rest/RestProxyUtils.java

```diff
@@ -95,12 +96,73 @@ public static Context mergeRequestOptionsContext(Context context, RequestOptions
         Context optionsContext = options.getContext();
 
         if (optionsContext != null && optionsContext != Context.NONE) {
-            context = Context.mergeContexts(context, optionsContext);
+            context = mergeContexts(context, optionsContext);
         }
 
         return context;
     }
 
+    /**
+     * Merges data from two {@link Context} instances.
+     *
+     * @param into The {@link Context} to merge data into.
+     * @param from The {@link Context} to merge data from.
+     *
+     * @return The merged {@link Context}.
+     */
+    private static Context mergeContexts(Context into, Context from) {
+        Objects.requireNonNull(into, "'into' cannot be null.");
+        Objects.requireNonNull(from, "'from' cannot be null.");
+
+        // If the 'into' Context is the NONE Context just return the 'from' Context.
+        // This is safe as Context is immutable and prevents needing to create any new Contexts and temporary arrays.
+        if (into == Context.NONE) {
+            return from;
+        }
+
+        // Same goes the other way, where if the 'from' Context is the NONE Context just return the 'into' Context.
+        if (from == Context.NONE) {
+            return into;
+        }
+
+        Context[] contextChain = getContextChain(from);
```

Comment:
We should add one additional case here, `from.getContextCount() == 1`. This will remove the creation of the `Context[]` and would just be

```java
if (from.getContextCount() == 1) {
    return into.addData(from.getKey(), from.getValue());
}

// Current code
```